src/unsafe_string.rs

```diff
@@ -276,10 +283,36 @@ impl IString {
         k.upgrade()
     }
 
+    /// Create an inline string by storing bytes in upper bits with null termination
+    /// Safety: String must be < 8 bytes and valid UTF-8
+    unsafe fn new_inline_string(s: &str) -> Self {
+        // 1 byte for the tag(3 bits for tag and rest for the length), 7 bytes for the string
+        let bytes = s.as_bytes();
+        let mut data: usize = 0;
+        data |= s.len() << TAG_SIZE_BITS;
+
+        for (i, &byte) in bytes.iter().enumerate() {
+            data |= (byte as usize) << (8 as usize + i * 8);
```

Comment:
Is byte order / endianness going to bite us? I think it might be preferable to `transmute` the data here.